paymentsheet/src/main/java/com/stripe/android/customersheet/data/CustomerSessionElementsSessionManager.kt

```diff
@@ -0,0 +1,98 @@
+package com.stripe.android.customersheet.data
+
+import com.stripe.android.core.injection.IOContext
+import com.stripe.android.customersheet.CustomerSheet
+import com.stripe.android.customersheet.ExperimentalCustomerSheetApi
+import com.stripe.android.model.ElementsSession
+import com.stripe.android.paymentsheet.ExperimentalCustomerSessionApi
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.PrefsRepository
+import com.stripe.android.paymentsheet.model.SavedSelection
+import com.stripe.android.paymentsheet.repositories.ElementsSessionRepository
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+internal interface CustomerSessionElementsSessionManager {
+    suspend fun fetchCustomerSessionEphemeralKey(): Result<CachedCustomerEphemeralKey>
+
+    suspend fun fetchElementsSession(): Result<ElementsSession>
+}
```

Comment:
This manager backs a majority of functions within `CustomerSheet`:
- Fetching `elements/session`
- Fetching payment methods
- Fetching ephemeral key when removing or updating payment methods

What will be eventually backed by this:
- Saving and retrieving the saved selection

What won't be backed by this
- Retrieving setup intent client secret.